torch/testing/_internal/common_utils.py

```diff
@@ -3094,7 +3094,7 @@ def assertRaisesRegex(self, expected_exception, expected_regex, *args, **kwargs)
         # Checks whether the test is instantiated for a device type by testing
         # if the test class has defined the device_type attribute and,
         # if so, tests whether the instantiated device type is native or not
-        if hasattr(self, 'device_type') and self.device_type not in NATIVE_DEVICES:  # type: ignore[attr-defined]
+        if hasattr(self, 'device_type') and self.device_type not in NATIVE_DEVICES and self.device_type != "mps":  # type: ignore[attr-defined]
```

Comment:
This is very weird...
I wonder if that code should just be removed altogether...